src/OpenLoco/src/Map/MapGenerator/MapGenerator.cpp

```diff
@@ -749,10 +752,95 @@ namespace OpenLoco::World::MapGenerator
     // Example: 'Electricity Pylon' building object
     static void generateMiscBuildingType1(const BuildingObject* buildingObj, const size_t id)
     {
-        registers regs;
-        regs.ebp = X86Pointer(buildingObj);
-        regs.ebx = id;
-        call(0x0042E893, regs);
+        for (auto& industry : IndustryManager::industries())
+        {
+            auto* industryObj = ObjectManager::get<IndustryObject>(industry.objectId);
+            if (!industryObj->hasFlags(IndustryObjectFlags::requiresElectricityPylons))
+                continue;
+
+            auto numTownsLeft = 3;
+            for (auto& town : TownManager::towns())
+            {
+                // Figure out distance and rotation (angle) between town and industry
+                auto rotationBL = 2;
+                auto xDist = town.x - industry.x; // ax
+                if (xDist < 0)
+                {
+                    xDist = -xDist;
+                    rotationBL = 0;
+                }
+
+                auto rotationBH = 1;
+                auto yDist = town.y - industry.y; // dx
+                if (yDist < 0)
+                {
+                    yDist = -yDist;
+                    rotationBH = 3;
+                }
+
+                auto totalDist = xDist + yDist; // bp
+                if (totalDist < 800 || totalDist > 2240)
+                    continue;
+
+                // Figure out the preferred rotation and distance to work with
+                if (yDist > xDist)
+                {
+                    std::swap(xDist, yDist);
+                    std::swap(rotationBL, rotationBH);
+                }
+
+                // 0x0042E91B
+                for (auto attemptsLeft = 2; attemptsLeft > 0; attemptsLeft--)
```

Comment:
Isn't this doing two lines like this:

```
I ------ x ------- x ------- x
                             |
                             |
                             |
                             |
                             x
                             |
                             |
                             |
                             |
                             T
```
Where I is the industry and T is the town. It does whatever the longest one is first then it does the shorter leg.